pkg/operator/baremetal_pod.go

```diff
@@ -46,14 +56,48 @@ func setMariadbPassword() corev1.EnvVar {
 		ValueFrom: &corev1.EnvVarSource{
 			SecretKeyRef: &corev1.SecretKeySelector{
 				LocalObjectReference: corev1.LocalObjectReference{
-					Name: "mariadb-password",
+					Name: baremetalSecretName,
 				},
-				Key: "password",
+				Key: baremetalSecretKey,
 			},
 		},
 	}
 }
 
+func generateRandomPassword() string {
+	rand.Seed(time.Now().UnixNano())
+	chars := []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
+		"abcdefghijklmnopqrstuvwxyz" +
+		"0123456789")
+	length := 16
+	buf := make([]rune, length)
+	for i := range buf {
+		buf[i] = chars[rand.Intn(len(chars))]
+	}
+	return (string(buf))
+}
+
+func createMariadbPasswordSecret(client coreclientv1.SecretsGetter, config *OperatorConfig) error {
```

Comment:
@sadasu can we have a unit test for this func?

Reply:
@enxebre please look at baremetal_test.go for the unit tests.